File: src/main/scala/com/github/unchama/seichiassist/commands/legacy/MapCommand.java
package com.github.unchama.seichiassist.commands.legacy;

import com.github.unchama.seichiassist.SeichiAssist;
import org.bukkit.Location;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.command.TabExecutor;
import org.bukkit.entity.Player;

import java.util.List;

public final class MapCommand implements TabExecutor {
    @Override
    public boolean onCommand(final CommandSender commandSender, final Command command, final String s, final String[] args) {
        if (commandSender instanceof Player) {
            final Player who = (Player) commandSender;
            final Location where = who.getLocation();
            who.sendMessage("http://map-s" + SeichiAssist.seichiAssistConfig().getServerNum()
                    + ".minecraftserver.jp/?worldname=" + who.getWorld().getName()
                    + "&mapname=flat&zoom=2&x=" + where.getBlockX() + "&y=" + where.getBlockY() + "&z=" + where.getBlockZ());
            return true;
        }
        commandSender.sendMessage("このコマンドはゲーム内から実行してください。");
        return false;
    }

    @Override
    public List<String> onTabComplete(final CommandSender commandSender, final Command command, final String s, final String[] args) {
        return null;
    }
}

File: src/main/scala/com/github/unchama/seichiassist/commands/MapCommand.scala
package com.github.unchama.seichiassist.commands

import cats.effect.IO
import com.github.unchama.seichiassist.SeichiAssist
import com.github.unchama.seichiassist.commands.contextual.builder.BuilderTemplates
import com.github.unchama.targetedeffect.syntax._
import org.bukkit.ChatColor._
import org.bukkit.command.TabExecutor

object MapCommand {
  val executor: TabExecutor = BuilderTemplates.playerCommandBuilder
    .execution { context =>
      IO {
        val location = context.sender.getLocation
        val url =
          s"$RED${UNDERLINE}http://map-s${SeichiAssist.seichiAssistConfig.getServerNum}.minecraftserver.jp" +
            s"/?worldname=${location.getWorld.getName}&mapname=flat&zoom=2&" +
            s"x=${location.getBlockX}&y=${location.getBlockY}&z=${location.getBlockZ}"
        url.asMessageEffect()
      }
    }
    .build()
    .asNonBlockingTabExecutor()
}
